softioc/extension.c

```diff
@@ -205,3 +200,60 @@ void EpicsPvPutHook(struct asTrapWriteMessage *pmessage, int after)
         /* Just save the old value for logging after. */
         pmessage->userPvt = value;
 }
+
+
+static PyObject *install_pv_logging(PyObject *self, PyObject *args)
+{
+    const char *acf_file;
+
+    if (!PyArg_ParseTuple(args, "s", &acf_file))
+        return NULL;
+
+    asSetFilename(acf_file);
+    asTrapWriteRegisterListener(EpicsPvPutHook);
+    Py_RETURN_NONE;
+}
+
+static struct PyMethodDef softioc_methods[] = {
+    {"get_DBF_values",  get_DBF_values, METH_VARARGS,
+     "Get a map of DBF names to values"},
+    {"get_field_offsets",  get_field_offsets, METH_VARARGS,
+     "Get offset, size and type for each record field"},
+    {"db_put_field",  db_put_field, METH_VARARGS,
+     "Put a database field to a value"},
+    {"install_pv_logging",  install_pv_logging, METH_VARARGS,
+     "Install caput logging to stdout"},
+    {NULL, NULL, 0, NULL}        /* Sentinel */
+};
+
+#if PY_MAJOR_VERSION >= 3
+static struct PyModuleDef softioc_module = {
+  PyModuleDef_HEAD_INIT,
+    "softioc._extension",
+    NULL,
+    -1,
+    softioc_methods,
+};
+#endif
+
+#if PY_MAJOR_VERSION >= 3
+#  define PyMOD(NAME) PyObject* PyInit_##NAME (void)
+#else
+#  define PyMOD(NAME) void init##NAME (void)
+#endif
+
+PyMOD(_extension)
+{
+#if PY_MAJOR_VERSION >= 3
+        PyObject *mod = PyModule_Create(&softioc_module);
+#else
+        PyObject *mod = Py_InitModule("softioc._extension", softioc_methods);
+#endif
+        if(mod) {
+        }
+#if PY_MAJOR_VERSION >= 3
+    return mod;
+#else
+    (void)mod;
+#endif
```

Comment:
This is very strange looking code!  The indentation is messed up, there's an empty statement in the middle, and we have a repeated `#if` test.  I'd suggest rewriting this as:
```c
#if PY_MAJOR_VERSION >= 3
    return PyModule_Create(&softioc_module);
#else
    Py_InitModule("softioc._extension", softioc_methods);
#endif
```